app/models/spree/payment.rb

```diff
@@ -21,7 +21,7 @@ class Payment < ApplicationRecord
     belongs_to :payment_method, class_name: 'Spree::PaymentMethod'
 
     has_many :offsets, -> { where("source_type = 'Spree::Payment' AND amount < 0").completed },
-             class_name: "Spree::Payment", foreign_key: :source_id
+             class_name: "Spree::Payment", foreign_key: :source_id, dependent: :nullify
```

Comment:
How confusing is this. It looks like there are payments which have another payment as source. They call it an offset here. I guess that it's the same as a (partial) refund.

This is now saying that if we delete a payment, we are keeping the its offsets. Same as before but it doesn't make much sense to me. Maybe good to make it explicit now though and not reference a non-existing record.

Reply:
keeping